experimenter/query.py

```diff
@@ -0,0 +1,147 @@
+from elasticsearch import Elasticsearch
+from elasticsearch_dsl import Search, Q
+from experimenter.utils import get_problem_path, get_dataset_doc_path, get_pipelines_from_d3m
+from d3m.runtime import _get_data_and_scoring_params_from_pipeline_run as _data_score_params
+
+
+HOST = 'https://metalearning.datadrivendiscovery.org/es'
+CONNECTION = Elasticsearch(hosts=[HOST], timeout=300)
+
+         
+def get_search_query(arguments: dict = None, connection = CONNECTION, index='pipelines'):
+    index_search = Search(using=CONNECTION, index=index)
+    if arguments['id'] is not None:
+        index_search = index_search.query('match', id=arguments['id'])
+    if arguments['submitter'] is not None:
+        index_search = index_search.query('match', _submitter=arguments['submitter'])
+    return index_search
+
+
+def query_on_seeds(pipeline_id: str=None, limit: int=None, submitter: str='byu'):
+    arguments = {'id': pipeline_id, 'submitter': submitter}
+    pipeline_search = get_search_query(arguments=arguments, index='pipelines')
+    for pipeline in pipeline_search.scan():
+        results = scan_pipeline_runs(pipeline.id, submitter)
+        for dataset_prob_tuple, results_dict in results.items():
+            unique_items = get_unique_results(results_dict)
+            #unpack values from tuple
+            problem_id, dataset_id, data_prep, scoring = dataset_prob_tuple
+            scoring_id, scoring_random_seed = scoring
+            data_prep_id, data_prep_seed = data_prep
+            #get preparation and scoring pipelines
+            data_prep_pipeline = get_pipeline(data_prep_id, types='Data')
+            scoring_pipeline = get_pipeline(scoring_id, types='Scoring')
+            for params in unique_items:
+                data_params = params['data_params']
+                scoring_params = params['scoring_params']
```

Comment:
There are a lot of loops here and some of the variable names are vague. It is hard for me to read what is happening in this function. Try breaking it up into smaller functions and use specific names.